GooglePlayInstant/LoadingScreen/LoadingBar.cs

```diff
@@ -117,23 +95,63 @@ private static void SetLoadingBarFill(GameObject loadingBarGameObject)
             var loadingBarFillImage = loadingBarFillGameObject.GetComponent<Image>();
             loadingBarFillImage.color = Color.green;
 
-            // Set size of component
-            var loadingBarFillRectTransform = loadingBarFillGameObject.GetComponent<RectTransform>();
-
-            var loadingBarRectTransform = loadingBarGameObject.GetComponent<RectTransform>();
-            loadingBarFillRectTransform.sizeDelta = new Vector2(0,
-                loadingBarRectTransform.sizeDelta.y - LoadingBarFillPadding);
-
             // Position fill component
             loadingBarFillGameObject.transform.position = loadingBarGameObject.transform.position;
         }
 #endif
 
+        /// <summary>
+        /// Sets the position of the loading bar as a function of the screen height and LoadingBarYAxisPercentage constant.
+        /// </summary>
+        public static void SetPosition()
+        {
+            var loadingBarRectTransform = GameObject.Find(LoadingBarGameObjectName).GetComponent<RectTransform>();
+
+            loadingBarRectTransform.position =
+                new Vector2(Screen.width / 2f, Screen.height * LoadingBarYAxisPercentage);
+        }
+
+        /// <summary>
+        /// Sets the length of the loading bar and its children as a function of the screen width and LoadingBarWidthPercentage constant.
+        /// </summary>
+        public static void SetWidth()
+        {
+            var loadingBarRectTransform = GameObject.Find(LoadingBarGameObjectName).GetComponent<RectTransform>();
+            var loadingBarFillRectTransform =
+                GameObject.Find(LoadingBarFillGameObjectName).GetComponent<RectTransform>();
+            var loadingBarOutlineRectTransform =
+                GameObject.Find(LoadingBarOutlineGameObjectName).GetComponent<RectTransform>();
+
+            loadingBarRectTransform.sizeDelta =
+                new Vector2(Screen.width * LoadingBarWidthPercentage,
+                    LoadingBarHeight);
+
+            loadingBarOutlineRectTransform.sizeDelta =
+                new Vector2(loadingBarRectTransform.sizeDelta.x,
```

Comment:
sizeDelta is a Vector2. Can you set it directly ?
`loadingBarOutlineRectTransform.sizeDelta = loadingBarRectTransform.sizeDelta;`